### ISSUE TITLE
go test results in error: invalid frame checksum

### ISSUE THREAD

**Issue by laurikoobas** [pierrec/lz4#29]
To reproduce I created a larger file:
> for i in {1..100};do cat Mark.Twain-Tom.Sawyer.txt >> longer.txt; done

And added the "longer.txt" into the _test files.

Result:
testdata/longer.txt                     : 24394635 / 18475893 / 38785100
--- FAIL: TestWriter (0.00s)
    --- FAIL: TestWriter/testdata/longer.txt/lz4.Header{CompressionLevel:_10_} (3.72s)
        writer_test.go:65: lz4: invalid frame checksum: got cdbda0bd; expected f1d62b24
FAIL
exit status 1
FAIL    github.com/pierrec/lz4  18.054s

I believe it's got something to do with io.Reader and it's quirk of not returning everything each time.

**Comment by pierrec** [comment]
Ah thanks for the report. I believe that this is a bug in the CompressBlockHC() function.
To be investigated.

**Comment by shenwei356** [comment]
Same issue here.

**Comment by pierrec** [comment]
I have confirmed it is a bug in CompressBlockHC(). Unfortunately, I have no time to fix it in the short term.
Either don't use high compression or send a pull request ;)!

**Comment by binary132** [comment]
From `writer.compressBlock`:

```go
	if level := z.Header.CompressionLevel; level != 0 {
		zn, err = CompressBlockHC(data, z.zdata, level)
	} else {
		zn, err = CompressBlock(data, z.zdata, z.hashtable[:])
	}
```

So this impacts all compression levels above 0, not just -9, right?

**Comment by pierrec** [comment]
Indeed. Sorry about that.

**Comment by pierrec** [comment]
Fixed as of v2.0.7
